### PR TITLE
Telos Cloud & WalletConnect login

### LINKED ISSUE #458: Migrate Antelope Wallets Authenticators to Teloscan
# Description
This is the last task in a series of tasks that has the ultimate goal of being able to authenticate users on Teloscan website using Telos Cloud and WalletConnect.

For this, we first need the refactoring of the Antelope wallets Authenticators to be able to handle custom functions as is a primary requirement on Teloscan.

Note: The correct implementation of this migration would include the export of the Antelope library outside the Telos Wallet project, which would imply a greater effort for which we do not have time at this moment. Therefore, the solution to this issue should be a workaround waiting for a final solution

# Aceptance Criteria

- users will be able to authenticate using WalletConnect and Telos Cloud
- users will be able to execute any function of any contract using the new authentication methods
- staking, unstaking and withdrawal will be possible to execute using the new authentication methods

### PR DESCRIPTION
**Pull request by Viterbo**
# Fixes #458

# Description
This PR adds the Telos Cloud option to authenticate the users.

# Test scenarios
You need to run this locally since the OAuth protocol does not work with Netlify links, only localhost or production domain.
```bash
git fetch origin
echo NETWORK=testnet > .env
git checkout 458-megrate-antelope-wallets-authenticators-to-teloscan
yarn; yarn dev
```
- go to http://localhost:8080
- Press "Sign in" button on top menu
- Select one of the options: Telos Cloud, WalletConnect, Metamask or Brave
  - a window popup should open. Login and create an account if you don't have one
  - verify you are logged in after that
- press F5 or refresh the page
  - you should still be logged without a window popping up

- go to Liquid Staking and try to:
  - Stake some tokens
  - Unstake some tokens
  - withdraw some unlocked tokens
  - all transactions should popup a confirmation window and the transaction should pass successfuly

- go to a known contract by contract address like: http://localhost:8080/address/0xa9991E4daA44922D00a78B6D986cDf628d46C4DD#contract
- go to contract section
  - you should see the list of functions
- try approve function
  - enter an address you control
  - enter any integer value
  - press the "Write" button
  - a new dialog should popup asking for confirmation
  - if you confirm, the transaction should pass successfully.

### CODE REVIEW

File: src/components/LoginModal.vue [hunk 1]
```diff
@@ -237,9 +290,9 @@ export default {
                 return accessGranted[0];
             }
         },
-        async ensureCorrectChain(checkProvider) {
+        async ensureCorrectChain(checkProvider: ethers.providers.Web3Provider) {
             const { chainId } = await checkProvider.getNetwork();
-            if (chainId !== process.env.NETWORK_EVM_CHAIN_ID) {
+            if (+chainId !== +(process.env.NETWORK_EVM_CHAIN_ID ?? '40')) {
```
Comment (ezra-sg) [inline src/components/LoginModal.vue:295]: lets make '40' a constant somewhere rather than magic string

File: src/components/LoginModal.vue [hunk 2]
```diff
@@ -263,7 +316,7 @@ export default {
             const provider = this.getInjectedProvider();
 
             if (provider) {
-                const chainId = parseInt(process.env.NETWORK_EVM_CHAIN_ID, 10);
+                const chainId = parseInt(process.env.NETWORK_EVM_CHAIN_ID || '40', 10);
```
Comment (ezra-sg) [inline src/components/LoginModal.vue:319]: same here

File: src/pages/staking/StakeForm.vue [hunk 3]
```diff
@@ -273,31 +285,64 @@ export default {
             this.displayConfirmModal = true;
         },
         initiateDeposit() {
+            console.log('initiateDeposit()');
```
Comment (ezra-sg) [inline src/pages/staking/StakeForm.vue:288]: remove log

File: src/pages/staking/StakeForm.vue [hunk 4]
```diff
@@ -273,31 +285,64 @@ export default {
...
+            const authenticator = useAccountStore().getAccount(CURRENT_CONTEXT).authenticator;
+
+            return authenticator.signCustomTransaction(
+                this.stlosContractInstance.address,
+                this.abi,
+                [],
+                value,
+            );
+        },
+        continueDemositLegacy(value: BigNumber) {
```
Comment (ezra-sg) [inline src/pages/staking/StakeForm.vue:338]: `continueDepositLegacy`

File: src/pages/staking/Staking.vue [hunk 5]
```diff
@@ -2,7 +2,15 @@
...
 import { mapGetters } from 'vuex';
 import { BigNumber, ethers } from 'ethers';
-import { formatWei, getRouteWatcherForTabs, WEI_PRECISION } from 'src/lib/utils';
+import {
+    formatWei,
+    getRouteWatcherForTabs,
+    LOGIN_DATA_KEY,
+    PROVIDER_WEB3_INJECTED,
+    // LOGIN_DATA_KEY,
+    // PROVIDER_WEB3_INJECTED,
```
Comment (ezra-sg) [inline src/pages/staking/Staking.vue:11]: remove commented lines

File: src/pages/staking/StakeForm.vue (hunk 4 above)
Comment (Viterbo) [reply src/pages/staking/StakeForm.vue:338]: In this case, I will argue that we need (at least for now) the two implementations living together. After we can replace that completely with the Antelope lib solution, then we can remove that part.

We can use another name if that's what you mean.

Comment (Viterbo) [reply src/pages/staking/StakeForm.vue:337]: OMG!!!!
I just realized what you meant. There was a typo and I argued about the legacy word LOL :rofl: 
Thanks, got it fixed

**Review by pmjanus — DISMISSED** [review]

**Review by pmjanus — APPROVED** [review]

**Review by donnyquixotic — APPROVED** [review]

**Pull request merged**